File: HomeWork/HomeWork2/HomeWork2.java
package HomeWork.HomeWork2;

public class HomeWork2{
    public static void main(String[] args){
        System.out.println(firstBadVersion(10));
    }
    public static int firstBadVersion(int n) {
        int low = 0;
        int high = n;
        while(true){
            int mid = low + ( high - low ) / 2;
             if(isBadVersion(mid) && !isBadVersion(mid-1)){
                 System.out.println("found");
                 return mid;
                 
             }else if(!isBadVersion(mid)){
                 System.out.println("oo low" + mid);
                 low = mid+1;
             }else{
                 System.out.println("too high" + mid);
                 high = mid;
             }
        }

     }
     public static boolean isBadVersion(int num){
        // 6 represents first bad version
        return num >= 6;
     }
}

File: HomeWork/HomeWork2/HomeWork2.py
def main():
   print(firstBadVersion(10))
    
def firstBadVersion(n):
    low = 1
    high = n
    while True:
        mid = low + (high - low) // 2
        if isBadVersion(mid) and not isBadVersion(mid - 1):
            return mid
        elif not isBadVersion(mid):
            low = mid + 1
        else:
            high = mid
            
def isBadVersion(s):
    return s >= 6

if __name__ == "__main__":
    main()